### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by robertroeser**
This pull request simplifies RSocket. It uses a single factory called RSocketFactory, with a type-safe builder.

Initiating a connection to a remote RSocket looks like this:
```
RSocket client = RSocketFactory
            .connect()
            .transport(TcpClientTransport.create("localhost", 7878))
            .start()
            .block();
```

Receiving connections from a remote RSocket looks like this:
```
RSocketFactory
            .receive()
            .acceptor(new PingHandler())
            .transport(TcpServerTransport.create(7878))
            .start()
            .subscribe();
```

This pull request currently removes Leases and Aeron support from the Java code base. Leases will be added back in shortly.

Sorry for the large request - setup was used in many places.

### CODE REVIEW

**Comment by robertroeser** [comment]
@benjchristensen @rdegnan @yschimke 
can't seem to add reviewers...can you guys take a look?

File: rsocket-core/src/main/java/io/rsocket/RSocketClient.java [hunk 1]
```diff
@@ -35,44 +32,96 @@
...
-        this.errorConsumer = new KnownErrorFilter(errorConsumer);
+        this.errorConsumer = errorConsumer;
         this.streamIdSupplier = streamIdSupplier;
-        this.keepAliveProvider = keepAliveProvider;
         this.started = MonoProcessor.create();
         this.senders = new IntObjectHashMap<>(256, 0.9f);
         this.receivers = new IntObjectHashMap<>(256, 0.9f);
+        this.missedAckCounter = new AtomicInteger();
+
+        if (tickPeriod != Duration.ZERO) {
```
Comment (yschimke) [inline rsocket-core/src/main/java/io/rsocket/RSocketClient.java]: A straight object reference comparison seems unsafe here.  

 * This is a <a href="{@docRoot}/java/lang/doc-files/ValueBased.html">value-based</a>
 * class; use of identity-sensitive operations (including reference equality
 * ({@code ==}), identity hash code, or synchronization) on instances of
 * {@code Duration} may have unpredictable results and should be avoided.
 * The {@code equals} method should be used for comparisons.

File: rsocket-core/src/main/java/io/rsocket/RSocketFactory.java [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+
+        T keepAliveAckTimeout(Duration ackTimeout);
+
+        T keepAliveMissedAcks(int missedAcks);
+    }
+
+    interface MimeType<T> {
+        T mimeType(String dataMineType, String metadataMimeType);
+
+        T dataMineType(String dataMineType);
```
Comment (yschimke) [inline rsocket-core/src/main/java/io/rsocket/RSocketFactory.java]: typo: dataMineType in a few places

File: rsocket-core/src/main/java/io/rsocket/RSocketFactory.java [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+        Fragmentation<ClientRSocketFactory, ClientTransport, RSocket, RSocket>,
+        ErrorConsumer<ClientRSocketFactory, ClientTransport, RSocket, RSocket>,
+        SetupPayload<ClientRSocketFactory> {
+
+        private Supplier<RSocket> acceptor = () -> new AbstractRSocket() {
+        };
+        private Supplier<io.rsocket.transport.ClientTransport> transportClient;
+        private Consumer<Throwable> errorConsumer = Throwable::printStackTrace;
+        private int mtu = 0;
+        private int flags = SetupFrameFlyweight.VALID_FLAGS - SetupFrameFlyweight.FLAGS_RESUME_ENABLE;
```
Comment (yschimke) [inline rsocket-core/src/main/java/io/rsocket/RSocketFactory.java]: Is it possible to disable leases anymore?

**Comment by yschimke** [comment]
Basic testing of the java cli is working, so no blocker from me.

https://github.com/yschimke/reactivesocket-cli/compare/rename...yschimke:rename2?expand=1

But this is too big to do much of a review in the PR.

File: rsocket-core/src/main/java/io/rsocket/RSocketServer.java [hunk 4]
```diff
@@ -40,49 +36,51 @@
...
-            .doFinally(signalType -> cleanup())
-            .subscribe();
-        if (requestHandler instanceof LeaseEnforcingSocket) {
-            LeaseEnforcingSocket enforcer = (LeaseEnforcingSocket) requestHandler;
-            enforcer.acceptLeaseSender(lease -> {
-                if (!clientHonorsLease) {
-                    return;
+                Collection<Subscription> values;
+                synchronized (this) {
+                    values = sendingSubscriptions.values();
```
Comment (yschimke) [inline rsocket-core/src/main/java/io/rsocket/RSocketServer.java]: IntObjectHashMap::values() looks like a live view of the map.  So I don't trust the concurrency here.

File: rsocket-core/src/main/java/io/rsocket/RSocketFactory.java (hunk 3 above)
Comment (robertroeser) [inline rsocket-core/src/main/java/io/rsocket/RSocketFactory.java]: @yschimke I removed leasing for the moment - trying to think of a way to add that back in cleaner manner than before. Should have added everything is disabled by default now, leasing, fragmentation, keep-alive, etc. You have to explicitly enable them.

Comment (yschimke) [inline rsocket-core/src/main/java/io/rsocket/RSocketFactory.java]: Are the flags right for leasing then, as I read it it indicates that it does honour leasing.

Comment (yschimke) [inline rsocket-core/src/main/java/io/rsocket/RSocketFactory.java]: Can you document the disabled functionality in the commit when it lands?

File: README.md [hunk 5]
```diff
@@ -15,7 +15,7 @@ Learn more at http://rsocket.io
 
 ## Build and Binaries
 
-<a href='https://travis-ci.org/RSocket/reactivesocket-java/builds'><img src='https://travis-ci.org/RSocket/reactivesocket-java.svg?branch=1.0.x'></a>
+<a href='https://travis-ci.org/ReactiveSocket/reactivesocket-java/builds'><img src='https://travis-ci.org/ReactiveSocket/reactivesocket-java.svg?branch=1.0.x'></a>
```
Comment (yschimke) [inline README.md]: remove this change

File: rsocket-core/src/main/java/io/rsocket/RSocketServer.java [hunk 6]
```diff
@@ -250,7 +241,7 @@ private synchronized void cleanUpChannelProcessors() {
                 .map(payload -> Frame.PayloadFrame.from(streamId, FrameType.NEXT, payload))
                 .transform(frameFlux -> {
                     LimitableRequestPublisher<Frame> frames = LimitableRequestPublisher.wrap(frameFlux);
-                    synchronized (this) {
+                    synchronized (RSocketServer.this) {
```
Comment (yschimke) [inline rsocket-core/src/main/java/io/rsocket/RSocketServer.java]: I suggest replacing the 3 lines with addSubscription

**Comment by yschimke** [comment]
👍🏾 from me for landing and iterating on this in smaller chunks in future 😀

**Comment by lexs** [comment]
LGTM for me too, I didn't look very closely at any code but the filenames look legit (no removed tests etc).

**Comment by robertroeser** [comment]
@yschimke @lexs if you guys are good with this can you merge it?

**Pull request merged**